### PR TITLE
[Filter/SharedModel] fix svace

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Fix svace issue,
- set next node when releasing the list item
- use macro to define local lock for shared table

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Review by dongju-chae — APPROVED** [review]
LGTM

**Review by gichan-jang — APPROVED** [review]
LGTM

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c [hunk 1]
```diff
@@ -2942,33 +2958,34 @@ nnstreamer_filter_shared_model_remove (void *instance, const char *key,
...
   model_rep->referred_list = g_list_remove (model_rep->referred_list, instance);
   ml_logd ("The referred instance of sharing key: %s has been removed!", key);
+  ret = TRUE;
 
   /* remove key from table if list is empty */
   if (g_list_length (model_rep->referred_list) == 0) {
     if (free_callback)
       free_callback (model_rep->shared_interpreter);
-    ret = g_hash_table_remove (shared_model_table, key);
+    g_hash_table_remove (shared_model_table, key);
```
Comment (zhoonit) [inline gst/nnstreamer/tensor_filter/tensor_filter_common.c:2985]: minor though, how about checking if either comment or the implementation should be changed?
`@return TRUE if the new key and value are removed. FALSE if failed to remove it.`

**Review by zhoonit — APPROVED** [review]
LGTM :)

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c (hunk 1 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_filter/tensor_filter_common.c:2985]: updated

**Review by again4you — APPROVED** [review]
LGTM!

**Pull request merged**